Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -3854,4 +3854,26 @@ Vector<JS::NonnullGCPtr<Element>> Document::elements_from_point(double x, double
     return sequence;
 }
 
+// https://drafts.csswg.org/cssom-view/#dom-document-scrollingelement
+JS::GCPtr<Element const> Document::scrolling_element() const
+{
+    // 1. If the Document is in quirks mode, follow these substeps:
+    if (in_quirks_mode()) {
+        // 1. If the body element exists, and it is not potentially scrollable, return the body element and abort these steps.
+        //    For this purpose, a value of overflow:clip on the the body element’s parent element must be treated as overflow:hidden.
+        if (auto const* body_element = body(); body_element && !body_element->is_potentially_scrollable())
+            return body_element;
+
+        // 2. Return null and abort these steps.
+        return nullptr;
+    }
+
+    // 2. If there is a root element, return the root element and abort these steps.
+    if (auto const* root_element = root().first_child_of_type<Element>(); root_element)
```

Comment:
I think this is supposed to be:
```c++
if (auto const* root_element = document_element())
```

Reply:
See: https://github.com/SerenityOS/serenity/pull/23143#discussion_r1484833673

Reply:
Yeah, both do the same in this case. I hadn't seen your comment before now, I should probably fix up the place that Andrew initially copied from :sweat_smile:

Reply:
At this point we should probably just add a function that implements https://drafts.csswg.org/css-display-4/#root-element and use that everywhere. :yakstack++: